src/providers/create-new/utils/buildProviderResources.ts

```diff
@@ -0,0 +1,43 @@
+import type { IoK8sApiCoreV1Secret, V1beta1Provider } from '@kubev2v/types';
+
+import { ProviderFormFieldId } from '../fields/constants';
+import type { CreateProviderFormData } from '../types';
+
+export const buildOvaProviderResources = (
+  formData: CreateProviderFormData,
+): {
+  provider: V1beta1Provider;
+  secret: IoK8sApiCoreV1Secret;
+} => {
+  const namespace = formData[ProviderFormFieldId.ProviderProject];
+  const providerName = formData[ProviderFormFieldId.ProviderName];
+  const nfsDirectory = formData[ProviderFormFieldId.NfsDirectory] ?? '';
+
+  const provider: V1beta1Provider = {
+    apiVersion: 'forklift.konveyor.io/v1beta1',
+    kind: 'Provider',
+    metadata: {
+      name: providerName,
+      namespace,
+    },
+    spec: {
+      secret: {
+        name: undefined,
+        namespace: undefined,
```

Comment:
here the namespace is undefined but you give a namespace to the secret below, doesn't the secret always gets created in the same namespace as the provider?

Reply:
Good point. I can use the namespace here directly.